packages/gatsby-transformer-xml/src/gatsby-node.js

```diff
@@ -1,44 +1,198 @@
-const parseXml = require(`xml-parser`)
+const convert = require(`xml-js`)
 const _ = require(`lodash`)
 
-async function onCreateNode({
-  node,
-  actions,
-  loadNodeContent,
-  createNodeId,
-  createContentDigest,
-}) {
-  const { createNode, createParentChildLink } = actions
-
-  // We only care about XML content.
+async function onCreateNode(
+  { node, actions, loadNodeContent, createNodeId, createContentDigest },
+  pluginOptions
+) {
   if (![`application/xml`, `text/xml`].includes(node.internal.mediaType)) {
     return
   }
-  const rawXml = await loadNodeContent(node)
-  const parsedXml = parseXml(rawXml)
-  const nodeArray = parsedXml.root.children.map((obj, i) => {
-    if (obj.children) {
-      obj.xmlChildren = obj.children
-      delete obj.children
+
+  // set the default options.
+  const defaultOptions = {
+    compact: true,
+    declarationKey: `declaration`,
+    instructionKey: `instruction`,
+    attributesKey: `attributes`,
+    textKey: `text`,
+    cdataKey: `cdata`,
+    doctypeKey: `doctype`,
+    parentKey: `parent`,
+  }
+
+  const {
+    useElementNamesAsKeys,
+    nativeType,
+    trim,
+    ignoreDeclaration,
+    ignoreInstruction,
+    ignoreAttributes,
+    ignoreComment,
+    ignoreCdata,
+    ignoreDoctype,
+    ignoreText,
+  } = pluginOptions
+
+  // merge defaults with plugin options
+  const options = Object.assign({}, defaultOptions, {
+    nativeType: typeof nativeType !== `undefined` ? nativeType : false,
+    trim: typeof trim !== `undefined` ? trim : false,
+    ignoreDeclaration:
+      typeof ignoreDeclaration !== `undefined` ? ignoreDeclaration : true,
+    ignoreInstruction:
+      typeof ignoreInstruction !== `undefined` ? ignoreInstruction : false,
+    ignoreAttributes:
+      typeof ignoreAttributes !== `undefined` ? ignoreAttributes : false,
+    ignoreComment: typeof ignoreComment !== `undefined` ? ignoreComment : true,
+    ignoreCdata: typeof ignoreCdata !== `undefined` ? ignoreCdata : false,
+    ignoreDoctype: typeof ignoreDoctype !== `undefined` ? ignoreDoctype : false,
```

Comment:
It can refactored into a `defaultPluginOptions` ?

And then:
```javascript
Object.assign({}, defaultOptions, defaultPluginOptions, pluginOptions)
```

Reply:
Right, I don't quite follow why we have 2 sets of defaults and they are implemented in bit different way.

I guess this might be related to this note in README:

```
Only a subset of the [xml-js options](https://www.npmjs.com/package/xml-js#options-for-converting-xml-%E2%86%92-js-object--json) are available. Allowing for non-compact mode would currently break the implementation.
```

but this could also be done with some white-listing (for example using `lodash`'s `_.pick`) if that's the case